sender-stackdriver/src/main/java/zipkin2/reporter/stackdriver/StackdriverSender.java

```diff
@@ -73,18 +77,26 @@ public StackdriverSender build() {
     }
   }
 
+  static final ByteString SPAN_ID_PREFIX = ByteString.copyFromUtf8("/spans/");
+
   final Channel channel;
   final CallOptions callOptions;
-  final String projectId;
+  final ByteString projectName;
+  final ByteString traceIdPrefix;
   final boolean shutdownChannelOnClose;
-  final int projectIdFieldSize;
+  final int projectNameFieldSize;
+  final int spanNameSize;
+  final int spanNameFieldSize;
 
   StackdriverSender(Builder builder) {
     channel = builder.channel;
     callOptions = builder.callOptions;
-    projectId = builder.projectId;
+    projectName = ByteString.copyFromUtf8("projects/" + builder.projectId);
+    traceIdPrefix = projectName.concat(ByteString.copyFromUtf8("/traces/"));
     shutdownChannelOnClose = builder.shutdownChannelOnClose;
-    projectIdFieldSize = 1 /* field no */ + CodedOutputStream.computeStringSizeNoTag(projectId);
+    projectNameFieldSize = 1 /* field no */ + CodedOutputStream.computeBytesSizeNoTag(projectName);
+    spanNameSize = traceIdPrefix.size() + 32 + SPAN_ID_PREFIX.size() + 16;
```

Comment:
the field is named spanNameXX but it seems it is something else. Is it spanIdsSize?